provider/datagen/src/bin/datagen.rs

```diff
@@ -197,38 +184,28 @@ fn main() -> eyre::Result<()> {
                 .takes_value(true)
                 .help(
                     "Include this locale in the output. Accepts multiple arguments. \
-                    Omit this option to include all locales.",
+                    Omit this option to include all locales. Set to 'none' for no locales.",
```

Comment:
### Discussion on `--all-locales` being default

Pros and cons for making this change:

1. Pro: Sets the default behavior to be the one that is best for i18n correctness.
2. Con: Makes the locale setting less discoverable.

Reply:
I don't think discoverability will be a problem, this is the main parameter that clients will expect.